### PR TITLE
add ClientPredictionStrategy

### PR DESCRIPTION
**Pull request by namel**

### CODE REVIEW

File: src/serialize/DynamicObject.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    }
+
+    step(worldSettings){
+        if (this.isRotatingRight){ this.angle += this.rotationSpeed; }
+        if (this.isRotatingLeft){this.angle -= this.rotationSpeed; }
+
+        if(this.angle>360){ this.angle -= 360; }
+        if(this.angle<0){ this.angle += 360; }
+
+        if (this.isAccelerating) {
```
Comment (OpherV) [inline src/serialize/DynamicObject.js]: This is spaaace-specific behavior

File: src/serialize/DynamicObject.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        this.velX = this.velocity.x;
+        this.velY = this.velocity.y;
+
+        this.isAccelerating = false;
+        this.isRotatingLeft = false;
+        this.isRotatingRight = false;
+        this.x = this.x + this.velocity.x;
+        this.y = this.y + this.velocity.y;
+
+        if (this.x>=worldSettings.width){ this.x = worldSettings.width - this.x;}
```
Comment (OpherV) [inline src/serialize/DynamicObject.js]: This is spaaace-specific behavior

File: src/serialize/DynamicObject.js (hunk 1 above)
Comment (namel) [inline src/serialize/DynamicObject.js]: Is it all spaaace-specific? Can we carve out some part which is generic for an object in two dimensional space, with an orientation, acceleration and deceleration constants, etc. ?

Comment (OpherV) [inline src/serialize/DynamicObject.js]: For pulses, acceleration, etc - maybe. 2d screen wrap - no.
Also maybe this delves too much into implementing a physics engine

Also the reason why this was in Spaaace to begin with is that I wanted an understandable, self contained example

**Pull request merged**